### PR TITLE
feature/SOF-4429 - Automated label encoding for categoricals

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: assets/python/ml/templates/model.pyi [hunk 1]
```diff
@@ -111,5 +111,11 @@ with settings.context as context:
         # Make some predictions
         predictions = model.predict(descriptors)
 
+        {% if category == "classification" %}
+        # Transform predictions back to their original labels
+        label_encoder: sklearn.preprocessing.LabelEncoder = context.load("label_encoder")
```
Comment (timurbazhirov) [inline assets/python/ml/templates/model.pyi:116]: Not sure I understand what this line actually does - pls explain? The ":" notation before "=" specifically.

Comment (AcylSilane) [reply assets/python/ml/templates/model.pyi:116]: It's type-hinting notation, to help the IDE with auto-completion. The general syntax goes:

`some_variable : type_of_the_variable = some_expression`

It's useful for cases like this, where `context.load` can return many different types, but we know it returns a LabelEncoder here.

**Comment by AcylSilane** [comment]
Merging per https://exabyte.atlassian.net/browse/SOF-4429?focusedCommentId=41435
